socorro/processor/rules/breakpad.py

```diff
@@ -70,6 +70,70 @@ def action(self, raw_crash, dumps, processed_crash, status):
         )
 
 
+class TruncateStacksRule(Rule):
+    """Truncate stacks that are too large
+
+    This rule truncates stacks that are greater than MAX_FRAMES. It truncates the stacks
+    in the middle with the thinking that the beginning and end of the stack are the most
+    interesting.
+
+    """
+
+    MAX_FRAMES = 500
+    HALF_MAX_FRAMES = int(MAX_FRAMES / 2)
+
+    def __init__(self):
+        super().__init__()
+        self.metrics = markus.get_metrics("processor.truncatestacksrule")
+
+    def truncation_frame(self, truncated_frames):
+        return {"truncated": {"msg": f"{len(truncated_frames):,} frames truncated"}}
+
+    def truncate(self, frames):
+        """Truncates a single stack if it's too big
+
+        :arg frames: the list of frame structures to truncate
+
+        :returns: truncated list of frame structures with a truncation
+            frame in the middle
+
+        """
+        self.metrics.gauge("stack_size", len(frames))
+        self.metrics.incr("truncated")
```

Comment:
This emits a `stack_size` metric so we can see the stack sizes over time. This only emits a metric for stacks > `MAX_FRAMES`. I think that's fine for now, but maybe at some point, we change it to emit a metric for all stacks.

This emits a `truncated` metric so we can see how often we truncate stacks.